### ISSUE TITLE
[E87] Fix Backward Pass Bugs Found by PyTorch Parity

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#645]
### Tasks

- [x] T87.1.1 Fix LayerNorm backward ReduceSum axis  Est: 30m  verifies: [UC-L01]  DONE 2026-04-10
- [x] T87.1.2 Run go test ./layers/normalization/... and TestParity_LayerNorm_Backward  Est: 15m  verifies: [in...
- [x] T87.2.1 Fix MatMul backward transposes  Est: 30m  verifies: [UC-L01]  DONE 2026-04-10
- [x] T87.2.2 Run go test ./layers/core/... and TestParity_MatMul_Backward  Est: 15m  verifies: [infrastructure...
- [x] T87.3.1 Fix MSE backward 2/N scaling  Est: 30m  verifies: [UC-L01]  DONE 2026-04-10
- [x] T87.3.2 Run go test ./training/loss/... and TestParity_MSELoss_Backward  Est: 15m  verifies: [infrastruct...
- [x] T87.4.1 Fix CrossEntropy backward 1/N normalization  Est: 30m  verifies: [UC-L01]  DONE 2026-04-10
- [x] T87.4.2 Run go test ./training/loss/... and TestParity_CrossEntropyLoss_Backward  Est: 15m  verifies: [in...

---
Synced from [plan.md](https://github.com/zerfoo/zerfoo/blob/main/docs/plan.md).

**Comment by dndungu** [comment]
Closing per ADR-093 tracker-hygiene pass (T128.5 follow-up): all member tasks complete (E61's last open item #727 was superseded by the standing DGX validation job, T131.x).
